pymongo/asynchronous/pool.py

```diff
@@ -860,8 +861,13 @@ async def _reset(
         # PoolClosedEvent but that reset() SHOULD close sockets *after*
         # publishing the PoolClearedEvent.
         if close:
-            for conn in sockets:
-                await conn.close_conn(ConnectionClosedReason.POOL_CLOSED)
+            if not _IS_SYNC:
+                await asyncio.gather(
```

Comment:
We probably want to use return_exceptions=True here to ensure all tasks complete.